src/compiler/sys.ts

```diff
@@ -869,6 +875,38 @@ namespace ts {
             };
             return nodeSystem;
 
+            /**
+             * Uses the builtin inspector APIs to capture a CPU profile
+             * See https://nodejs.org/api/inspector.html#inspector_example_usage for details
+             */
+            function enableCPUProfiler(path: string, cb: () => void) {

```

Comment:
What happens if this is called twice before `disableCPUProfiler`?  Does it matter?

Reply:
As-is, minimally only one (the last one) is going to be completed - however that's assuming the runtime even allow multiple sessions to connect simultaneously. I can add a check for an existing session and a return flag for if the profiler is already enabled

Reply:
Or just throw?

Reply:
Nah - given that it's supposed to be a relatively unobtrusive profile, I'd rather just not generate the profile data. If a caller wants to throw, they can inspect the return value and do so.

Reply:
In any case, it's now implemented as suggested in my prior comment - the function is effectively a passthru to the callback if a profiling run is already ongoing.